File: programers_Lv2/짝지어 제거하기.py
def solution(s):
    rst = []

    rst.append(s[0])

    for c in s[1:]:
        if not rst:
            rst.append(c)
        elif rst[-1] != c:
            rst.append(c)
        elif rst[-1] == c:
            del(rst[-1])

    return (lambda x: 1 if not rst else 0)(rst)

print(solution('baabaa'))
print(solution('cdcd'))

File: programers_Lv2/짝지어 제거하기.js
function solution(s) {
  const rst = [];

  rst[0] = s[0];
  ss = s.substring(1);

  for (let el of ss) {
    if (!rst) {
      rst.push(el);
    } else if (rst[rst.length - 1] !== el) {
      rst.push(el);
    } else if (rst[rst.length - 1] === el) {
      rst.pop();
    }
  }

  return rst.length === 0 ? 1 : 0;
}

console.log(solution("baabaa"));
console.log(solution("cdcd"));
